### PR TITLE
EVG-16236 revendor grip

### PR DESCRIPTION
**Pull request by ybrill**
[EVG-16236](https://jira.mongodb.org/browse/EVG-16236)

### Description 
* Update grip and jasper to accommodate the changes in https://github.com/mongodb/grip/commit/898ba2888109b89e382fe10b1aed1574fe9fe484.
* Add the buffer factor to admin settings
* Use the buffered async sender for Splunk messages.

### Testing 
In staging the admin setting propagated from the UI to the db. I put the dist up in staging and I'll check Spunk to see if the total message count changes at all. I'll also check the systemd logs for ["the message was dropped because the buffer was full"](https://github.com/mongodb/grip/blob/898ba2888109b89e382fe10b1aed1574fe9fe484/send/buffered_async.go#L107) and other errors.

### CODE REVIEW

File: config_logger.go [hunk 1]
```diff
@@ -79,15 +76,43 @@ func (c *LoggerConfig) ValidateAndDefault() error {
...
+	}
+
+	if b.Count < 0 {
+		catcher.Add(errors.New("buffering count can not be negative"))
+	} else if b.Count == 0 {
+		b.Count = defaultLogBufferingCount
+	}
+
+	if b.IncomingBufferFactor < 0 {
+		catcher.Add(errors.New("incoming buffer factor can not be negative"))
```
Comment (Kimchelly) [inline config_logger.go:112]: `catcher.New`

File: config_logger.go [hunk 2]
```diff
@@ -79,15 +76,43 @@ func (c *LoggerConfig) ValidateAndDefault() error {
 		c.ThresholdLevel = "debug"
 	}
 
+	catcher := grip.NewBasicCatcher()
+	catcher.Add(c.Buffer.validateAndDefault())
+
 	info := c.Info()
 	if !info.Valid() {
-		return errors.Errorf("logging level configuration is not valid [%+v]", info)
+		catcher.Add(errors.Errorf("logging level configuration is not valid [%+v]", info))
```
[the remark this answers is not in the document]
Comment (ybrill) [reply config_logger.go]: done

File: config_logger.go [hunk 3]
```diff
@@ -79,15 +76,43 @@ func (c *LoggerConfig) ValidateAndDefault() error {
...
+	catcher := grip.NewBasicCatcher()
+
+	if b.DurationSeconds < 0 {
+		catcher.Add(errors.New("buffering duration seconds can not be negative"))
+	} else if b.DurationSeconds == 0 {
+		b.DurationSeconds = defaultLogBufferingDuration
+	}
+
+	if b.Count < 0 {
+		catcher.Add(errors.New("buffering count can not be negative"))
```
[the remark this answers is not in the document]
Comment (ybrill) [reply config_logger.go]: done

File: config_logger.go [hunk 4]
```diff
@@ -79,15 +76,43 @@ func (c *LoggerConfig) ValidateAndDefault() error {
...
+	DurationSeconds      int `bson:"duration_seconds" json:"duration_seconds" yaml:"duration_seconds"`
+	Count                int `bson:"count" json:"count" yaml:"count"`
+	IncomingBufferFactor int `bson:"incoming_buffer_factor" json:"incoming_buffer_factor" yaml:"incoming_buffer_factor"`
+}
+
+func (b *LogBuffering) validateAndDefault() error {
+	catcher := grip.NewBasicCatcher()
+
+	if b.DurationSeconds < 0 {
+		catcher.Add(errors.New("buffering duration seconds can not be negative"))
```
[the remark this answers is not in the document]
Comment (ybrill) [reply config_logger.go]: done

File: config_logger.go (hunk 1 above)
Comment (ybrill) [reply config_logger.go]: done

**Comment by ybrill** [comment]
> I would test the REST model ([1](https://github.com/evergreen-ci/evergreen/blob/fe9e237b1b54b0f2a05c28394a90c9707da263f7/rest/model/admin_test.go#L132), [2](https://github.com/evergreen-ci/evergreen/blob/fe9e237b1b54b0f2a05c28394a90c9707da263f7/rest/model/admin_test.go#L215)), data connector ([1](https://github.com/evergreen-ci/evergreen/blob/fe9e237b1b54b0f2a05c28394a90c9707da263f7/rest/data/admin_test.go#L162), [2](https://github.com/evergreen-ci/evergreen/blob/fe9e237b1b54b0f2a05c28394a90c9707da263f7/rest/data/admin_test.go#L287)) and [route](https://github.com/evergreen-ci/evergreen/blob/fe9e237b1b54b0f2a05c28394a90c9707da263f7/rest/route/admin_test.go#L146) for the admin settings change.

done

**Review by Kimchelly — APPROVED** [review]
LGTM mod bump the agent version.

**Comment by ybrill** [comment]
> bump the agent version

done

**Comment by ybrill** [comment]
@Kimchelly Brian suggested that instead of merging this straight we should add an admin option to toggle it. 
Would you mind to look over the ["add option" commit](https://github.com/evergreen-ci/evergreen/pull/5393/commits/9a5839ce7dc139ec43341af3944ba96bde847c3f)?

File: config.go [hunk 5]
```diff
@@ -556,12 +556,23 @@ func (s *Settings) makeSplunkSender(ctx context.Context, client *http.Client, le
...
+	if s.LoggerConfig.Buffer.UseAsync {
+		if sender, err = send.NewBufferedAsyncSender(ctx,
+			sender,
+			send.BufferedAsyncSenderOptions{
+				BufferedSenderOptions: opts,
+				IncomingBufferFactor:  s.LoggerConfig.Buffer.IncomingBufferFactor,
+			}); err != nil {
+			return nil, errors.Wrap(err, "making splunk async buffered sender")
+		}
+	} else {
```
Comment (Kimchelly) [inline config.go:572]: nit L572-573 can be combined into an `else if`, instead of `else` then `if`.

**Review by Kimchelly — APPROVED** [review]
Admin setting also LGTM mod nit.

**Pull request merged**